validator/immutable_attribute_test.go

```diff
@@ -0,0 +1,65 @@
+package validator
+
+import (
+	"testing"
+)
+
+func Test_ValidateImmutableAttribute(t *testing.T) {
+	testCases := []struct {
+		Received       map[string]interface{}
+		Blacklist      map[string]interface{}
+		ErrorMatcher   func(err error) bool
+		ErrorAttribute string
+	}{
+		// With no black list provided, it shouldn't complain at all
+		{
+			Received:       map[string]interface{}{"name": "Oliver", "age": 29},
+			Blacklist:      map[string]interface{}{},
+			ErrorAttribute: "",
+			ErrorMatcher:   nil,
+		},
+
+		// With an attribute in the black list that is not present in the received
+		// struct, it shouldn't complain at all
+		{
+			Received:       map[string]interface{}{"name": "Oliver", "age": 29},
+			Blacklist:      map[string]interface{}{"hairColor": ""},
+			ErrorAttribute: "",
+			ErrorMatcher:   nil,
+		},
+
+		// With an attribute in the black list that _is_ present in the received
+		// struct, it should return a ImmutableAttribute error
+		{
+			Received:       map[string]interface{}{"name": "Oliver", "age": 29},
+			Blacklist:      map[string]interface{}{"name": ""},
```

Comment:
Good to see that in action. I wonder now why blacklist is a map as well. It looks to me the map values are always empty. Why is that?